cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -370,5 +396,188 @@ func loadApproverOptions(s *GCPControllerManager) (approverOptions, error) {
 	// Put master's zone first.
 	sort.Slice(a.zones, func(i, j int) bool { return a.zones[i] == zone })
 
+	a.clusterName, err = metadata.Get("instance/attributes/cluster-name")
```

Comment:
It would be nice to support running this in GCE too. Do hosted masters have this set?

Reply:
Hosted masters have this set.
Not sure how to support GCE clusters. Should we add a flag for this that disables cluster membership check?